From: Johannes Berg <johannes.berg@intel.com>
Date: Thu, 12 Mar 2020 16:59:13 +0100
Subject: backports: netlink: fix nla_validate_nested()

I had accidentally named this nl80211_validate_nested(),
and it got renamed in 5.6.

Signed-off-by: Johannes Berg <johannes.berg@intel.com>
(cherry picked from commit f1708ac5643a1decf36ca92ba0644397df5da1f9
https://git.kernel.org/pub/scm/linux/kernel/git/backports/backports.git)
Signed-off-by: You-Sheng Yang <vicamo.yang@canonical.com>
---
 backport-include/net/netlink.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/backport-include/net/netlink.h b/backport-include/net/netlink.h
index 9a5c270..942217e 100644
--- a/backport-include/net/netlink.h
+++ b/backport-include/net/netlink.h
@@ -266,17 +266,17 @@ nla_validate_nested_deprecated(const struct nlattr *start, int maxtype,
 }
 #endif /* < 5.2 */
 
-#if LINUX_VERSION_IS_LESS(5,3,0)
-#define nl80211_validate_nested LINUX_BACKPORT(nl80211_validate_nested)
+#if LINUX_VERSION_IS_LESS(5,6,0)
+#define nla_validate_nested LINUX_BACKPORT(nla_validate_nested)
 static inline int
-nl80211_validate_nested(const struct nlattr *start, int maxtype,
-			const struct nla_policy *policy,
-			struct netlink_ext_ack *extack)
+nla_validate_nested(const struct nlattr *start, int maxtype,
+		    const struct nla_policy *policy,
+		    struct netlink_ext_ack *extack)
 {
 	return __nla_validate_nested(start, maxtype, policy,
 				     NL_VALIDATE_STRICT, extack);
 }
-#endif /* < 5.3 */
+#endif /* < 5.6 */
 
 #if LINUX_VERSION_IS_LESS(5,1,0)
 #undef NLA_POLICY_NESTED
